Sources/Stevenson/JiraService.swift

```diff
@@ -130,24 +130,51 @@ extension JiraService {
 
 extension JiraService {
     public struct Version: Content {
-        public var id: String?
+        public let id: String?
         public let projectId: Int
-        public let description: String
+        public let description: String?
         public let name: String
         let released: Bool
-        @CustomCodable<YMDDate>
-        var startDate: Date
+        let startDate: CustomCodable<YMDDate>?
 
-        public init(projectId: Int, description: String, name: String, released: Bool = false, startDate: Date) {
-            self.id = nil
+        public init(id: String? = nil, projectId: Int, name: String, description: String?, released: Bool = false, startDate: Date?) {
+            self.id = id
             self.projectId = projectId
-            self.description = description
             self.name = name
+            self.description = description
             self.released = released
-            self.startDate = startDate
+            self.startDate = startDate.map(CustomCodable<YMDDate>.init(wrappedValue:))
         }
     }
 
+    public func getVersions(project projectID: Int, on container: Container) throws -> Future<[Version]> {
+        let fullURL = URL(string: "/rest/api/3/project/\(projectID)/versions", relativeTo: baseURL)!
+
+        let projectKey = self.knownProjects.first(where: { $0.value == projectID })?.key ?? "#\(projectID)"
+        let logMessage = "Fetching JIRA versions for board <\(projectKey)>"
+        self.logger.info("[JIRA] \(logMessage)")
```

Comment:
off-topic: could have a logger helper method to take care of prefix

Reply:
Fixed, indeed feels more in par with the other helpers already existing in this file anyway 👍